rust/arrow/examples/dynamic_types.rs

```diff
@@ -58,7 +58,8 @@ fn main() {
     ]);
 
     // build a record batch
-    let batch = RecordBatch::new(Arc::new(schema), vec![Arc::new(id), Arc::new(nested)]);
+    let batch =
+        RecordBatch::new(Arc::new(schema), vec![Arc::new(id), Arc::new(nested)]).unwrap();
```

Comment:
nit: you can replace `unwrap()` with `?` for better error message.

Reply:
The function doesn't return a Result, so I can't `?` there. Should I leave the unwrap?

Reply:
In `main` you can actually use `?`, see [here](https://github.com/rust-lang/rfcs/pull/1937).

Reply:
If my main has a `Result` return type? I tried it before commenting, and I got an error about needing to implement `TryFrom`.

I've got it working with returning `Result<()>`

Reply:
Yes you are right - seems you have to change the `main` interface and add return `Ok(())` from the main as well.. I guess it may not worth it for this purpose. We can just keep as it is.

Reply:
:( I had already changed it, can I keep it the way it is now? If I need to change anything else on the PR I'll revert it back to its original form

Reply:
OK. Let's keep as it is.